### PR TITLE
Add py27 and py37 to the build

### PR DESCRIPTION
**Pull request by bam241**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bam241** [comment]
@conda-forge-admin, please re-render

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ requirements:
     - openblas 0.2.20|0.2.20.*
     - lapack
     - boost-cpp 1.66.0
-    - hdf5 1.10.1
+    - hdf5
```
Comment (scopatz) [inline recipe/meta.yaml:34]: Actually, all of the pins here should be removed

**Comment by bam241** [comment]
@conda-forge-admin, please re-render

File: recipe/meta.yaml (hunk 1 above)
Comment (bam241) [reply recipe/meta.yaml:34]: the remaining ones are actually required...

Comment (scopatz) [reply recipe/meta.yaml:34]: Actually, I think they are now covered in conda-forge-pinning.  It is a mistake to keep these.

Comment (bam241) [reply recipe/meta.yaml:34]: I am not sure I understand, in a docker image, if I remove the 1.66 and the 2.48 for `boost` and `glibmm` I can't compile `cyclus`...

Comment (scopatz) [reply recipe/meta.yaml:34]: Oh really? That is bad...

Comment (bam241) [reply recipe/meta.yaml:34]: we depends on `boost/uuid/sha1.hpp` that has been deprecated, and we have some symbol issue with `glibmm`
This need to be fix, but as I don't have the time right now...

(edit: I put the wrong boost header... the issue is with `boost/uuid/sha1.hpp`)

Comment (scopatz) [reply recipe/meta.yaml:34]: Ahhh ok, that makes sense

**Comment by gonuke** [comment]
Thanks @bam241

**Pull request merged**